posthog/temporal/data_imports/external_data_job.py

```diff
@@ -0,0 +1,219 @@
+            start_to_close_timeout=dt.timedelta(minutes=1),
+            retry_policy=RetryPolicy(
+                initial_interval=dt.timedelta(seconds=10),
+                maximum_interval=dt.timedelta(seconds=60),
+                maximum_attempts=0,
+                non_retryable_error_types=["NotNullViolation", "IntegrityError"],
+            ),
+        )
+
+        update_inputs = UpdateExternalDataJobStatusInputs(
+            id=run_id, run_id=run_id, status=ExternalDataJob.Status.COMPLETED, latest_error=None
+        )
+
+        try:
+            # TODO: can make this a child workflow for separate worker pool
+            source_schemas = await workflow.execute_activity(
+                run_external_data_job,
+                inputs,
+                start_to_close_timeout=dt.timedelta(minutes=30),
+                retry_policy=RetryPolicy(maximum_attempts=3),
+                heartbeat_timeout=dt.timedelta(minutes=1),
+            )
+
+            # check schema first
+            validate_inputs = ValidateSchemaInputs(
+                source_schemas=source_schemas, external_data_source_id=inputs.external_data_source_id, create=False
+            )
+
+            await workflow.execute_activity(
+                validate_schema_activity,
+                validate_inputs,
+                start_to_close_timeout=dt.timedelta(minutes=2),
+                retry_policy=RetryPolicy(maximum_attempts=2),
+            )
+
+            # if not errors, then create the schema
+            validate_inputs.create = True
+            await workflow.execute_activity(
+                validate_schema_activity,
+                validate_inputs,
+                start_to_close_timeout=dt.timedelta(minutes=2),
+                retry_policy=RetryPolicy(maximum_attempts=2),
+            )
+
+            move_inputs = MoveDraftToProductionExternalDataJobInputs(
+                team_id=inputs.team_id,
+                external_data_source_id=inputs.external_data_source_id,
+            )
+
+            await workflow.execute_activity(
+                move_draft_to_production_activity,
+                move_inputs,
+                start_to_close_timeout=dt.timedelta(minutes=1),
+                retry_policy=RetryPolicy(maximum_attempts=2),
+            )
+
+        except exceptions.ActivityError as e:
+            if isinstance(e.cause, exceptions.CancelledError):
+                update_inputs.status = ExternalDataJob.Status.CANCELLED
+            else:
+                update_inputs.status = ExternalDataJob.Status.FAILED
+
+            update_inputs.latest_error = str(e.cause)
+            raise
+        except Exception as e:
+            if isinstance(e, SchemaValidationError):
+                update_inputs.latest_error = "Schema validation failed"
+            else:
+                update_inputs.latest_error = "An unexpected error has ocurred"
+
+            update_inputs.status = ExternalDataJob.Status.FAILED
+            raise
```

Comment:
nit: I'd recommend logging here for debugging purposes, but this can be added later.